### PR TITLE
Publishing a shared output

### PR DESCRIPTION
**Pull request by markcsims**

### CODE REVIEW

File: apps/crn-frontend/src/network/teams/state.ts [hunk 1]
```diff
@@ -165,3 +176,18 @@ export const useAuthorSuggestions = () => {
...
+  const authorization = useRecoilValue(authorizationState);
+  const setRefreshListing = useRefreshResearchOutputListing();
+  const setResearchOutputItem = useSetResearchOutputItem();
+  return async (payload: ResearchOutputPostRequest) => {
+    const researchOutput = await createTeamResearchOutput(
+      payload,
+      authorization,
+    );
+    setResearchOutputItem(researchOutput);
+    setRefreshListing();
```
Comment (kmaid) [inline apps/crn-frontend/src/network/teams/state.ts:190]: I think this is broken. Shouldn't it increase the integer?

File: apps/crn-frontend/src/shared-research/state.ts [hunk 2]
```diff
@@ -156,17 +154,13 @@ const setResearchOutput = selector<ResearchOutputResponse | undefined>({
   },
 });
 
-export const usePostTeamResearchOutput = () => {
-  const authorization = useRecoilValue(authorizationState);
+export const useRefreshResearchOutputListing = () => {
```
Comment (kmaid) [inline apps/crn-frontend/src/shared-research/state.ts:157]: I think if we increase the number from the other place we can remove this hook as it should never be called in isolation

File: apps/crn-frontend/src/shared-research/state.ts [hunk 3]
```diff
@@ -1,30 +1,42 @@
...
 const researchOutputIndexState = atomFamily<
   { ids: ReadonlyArray<string>; total: number } | Error | undefined,
-  ResearchOutputListOptions
+  RefreshResearchOutputListOptions
 >({
   key: 'researchOutputIndex',
   default: undefined,
 });
+
+const refreshResearchOutputListing = atom<number>({
```
Comment (kmaid) [inline apps/crn-frontend/src/shared-research/state.ts:35]: Perhaps this should be called refreshResearchOutputIndex?

File: packages/react-components/src/organisms/Form.tsx [hunk 4]
```diff
@@ -53,11 +51,12 @@ const Form = <T extends void | Record<string, unknown>>({
     if (formRef.current!.reportValidity() && parentValidation) {
       setStatus('isSaving');
       try {
-        const result = await onSave();
-        if (formRef.current) {
-          setStatus('hasSaved');
-        }
-        return result;
+        return await onSave().then((result) => {
```
Comment (kmaid) [inline packages/react-components/src/organisms/Form.tsx:54]: I think we can remove this promise chain to improve readability

File: packages/react-components/src/organisms/TeamCreateOutputForm.tsx [hunk 5]
```diff
@@ -160,9 +161,10 @@ const TeamCreateOutputForm: React.FC<TeamCreateOutputFormProps> = ({
         authors.length !== 0 ||
         identifierType !== ResearchOutputIdentifierType.None ||
         identifier !== '' ||
+        labCatalogNumber !== '' ||
         teams.length !== 1 // Original team
       }
-      onSave={() => {
+      onSave={async () => {
```
Comment (kmaid) [inline packages/react-components/src/organisms/TeamCreateOutputForm.tsx:167]: This isn't async, Do we want to remove this?

File: packages/react-components/src/organisms/__tests__/Form.test.tsx [hunk 6]
```diff
@@ -17,6 +17,7 @@ import { Button } from '../../atoms';
 const props: ComponentProps<typeof Form> = {
   dirty: false,
   children: () => null,
+  onSave: jest.fn(),
```
Comment (kmaid) [inline packages/react-components/src/organisms/__tests__/Form.test.tsx:20]: I think this should be a onSave: ()=>Promise.Resolve()

File: apps/crn-frontend/src/network/teams/state.ts (hunk 1 above)
Comment (markcsims) [reply apps/crn-frontend/src/network/teams/state.ts:190]: That's what setRefreshListing  is doing. Although we could add that call to setResearchOutputItem?

File: apps/crn-frontend/src/shared-research/state.ts (hunk 3 above)
Comment (markcsims) [reply apps/crn-frontend/src/shared-research/state.ts:35]: Done

File: apps/crn-frontend/src/network/teams/state.ts (hunk 1 above)
Comment (markcsims) [reply apps/crn-frontend/src/network/teams/state.ts:190]: Done as below

File: apps/crn-frontend/src/shared-research/state.ts (hunk 2 above)
Comment (markcsims) [reply apps/crn-frontend/src/shared-research/state.ts:157]: Done

File: packages/react-components/src/organisms/Form.tsx (hunk 4 above)
Comment (markcsims) [reply packages/react-components/src/organisms/Form.tsx:54]: Done

File: packages/react-components/src/organisms/TeamCreateOutputForm.tsx (hunk 5 above)
Comment (markcsims) [reply packages/react-components/src/organisms/TeamCreateOutputForm.tsx:167]: done

File: packages/react-components/src/organisms/__tests__/Form.test.tsx (hunk 6 above)
Comment (markcsims) [reply packages/react-components/src/organisms/__tests__/Form.test.tsx:20]: done

**Review by kmaid — APPROVED** [review]

**Pull request merged**